apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.ts

```diff
@@ -0,0 +1,100 @@
+import { Router } from 'express';
+import asyncHandler = require('express-async-handler');
+import error = require('@prairielearn/error');
+import { v4 as uuidv4 } from 'uuid';
+import { CourseSchema } from '../../lib/db-types';
+import { InstructorSharing } from './instructorCourseAdminSharing.html';
+import sqldb = require('@prairielearn/postgres');
+
+const router = Router();
+const sql = sqldb.loadSqlEquiv(__filename);
+
+async function generateSharingId(res) {
+  const newSharingId = uuidv4();
+  return await sqldb.queryZeroOrOneRowAsync(sql.update_sharing_token, {
+    sharing_token: newSharingId,
+    course_id: res.locals.course.id,
+  });
+}
+
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.question_sharing_enabled) {
+      throw error.make(403, 'Access denied (feature not available)');
+    }
+
+    const sharingInfo = await sqldb.queryRow(
+      sql.get_course_sharing_info,
+      {
+        course_id: res.locals.course.id,
+      },
+      CourseSchema,
+    );
+
+    if (!sharingInfo.sharing_token) {
+      await generateSharingId(res);
+      res.redirect(req.originalUrl);
+      return;
+    }
```

Comment:
If we added a default to this column and made it non-nullable, we could avoid this little dance.

Reply:
done